streaming/base/format/mds/writer.py

```diff
@@ -80,7 +80,7 @@ def encode_sample(self, sample: Dict[str, Any]) -> bytes:
                 size = len(datum)
                 sizes.append(size)
             else:
-                assert size == len(datum)
+                assert size == len(datum), f'Unexpected data size; was this data typed with the correct encoding ({encoding})?'
```

Comment:
Thanks for adding a message. Can you please also change the `assert` statement to `KeyError` Exception ? We would like to avoid having `assert` statement in the src code. Thanks!